### PR TITLE
Remove tester blockchain from the tests

### PR DESCRIPTION
**Pull request by LefterisJP**
We don't really use it, except when it breaks the tests. Time to get rid of it.

Opted to keep the blockchain-type argument in the tests in case we need
differentiation between different clients or support different kinds of
blockchains in the future.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -79,12 +79,6 @@ jobs:
         - SMOKETEST=1
         - RUN_SYNAPSE=1
 
-    - stage: test
-      <<: *job-template-test
-      env:
-        - BLOCKCHAIN_TYPE='tester'
```
Comment (palango) [inline .travis.yml]: I would like to keep this test and just change to `geth` here.

Comment (hackaugusto) [reply .travis.yml]: definitely this needs to be kept, it's the only stage that runs the unit tests

Comment (LefterisJP) [reply .travis.yml]: ops good point. Wanted to only remove tester contracts integration there. Amending now.

Comment (LefterisJP) [reply .travis.yml]: But I will remove the contracts tests. They run again at integration

**Comment by palango** [comment]
Rebased this on latest master

**Pull request merged**